src/util/invariant.h

```diff
@@ -227,13 +227,23 @@ inline void invariant_violated_string(
 // The condition should only contain (unmodified) arguments to the method.
 // "The design of the system means that the arguments to this method
 //  will always meet this condition".
+//
+// The condition asserts properties that the function expects, including,
+// member variables of the object, in case they are implicit input to
+// the function. It is also supposed to signal that even though an object
+// is in a valid state, a particular function may not be operating on
+// objects with the particular properties this one enjoys.
 #define PRECONDITION(CONDITION) INVARIANT(CONDITION, "Precondition")
 #define PRECONDITION_STRUCTURED(CONDITION, TYPENAME, ...) \
   INVARIANT_STRUCTURED(CONDITION, TYPENAME, __VA_ARGS__)
 
 // The condition should only contain variables that will be returned /
 // output without further modification.
 // "The implementation of this method means that the condition will hold".
+//
+// A POSTCONDITION documents what the function can guarantee after a "normal"
+// execution about the state it returns to its caller (properties of objects
+// that it returns).
```

Comment:
Can we rephrase this to something like: `A POSTCONDITION documents what the method guarantees about its output when it returns, given that it was called with valid inputs. Outputs include the return value of the method, as well as member variables that the method may write.`